bemani/backend/iidx/cannonballers.py

```diff
@@ -1,15 +1,2184 @@
+                matching_class_range.add_child(Node.s32('matching_class', matching_class))
+                matching_class_range.add_child(Node.s32('low_arena_class', 1))
+                matching_class_range.add_child(Node.s32('high_arena_class', 20))
+        return root
+
+    def handle_IIDX25pc_eaappliresult_request(self, request: Node) -> Node:
+        clear_map = {
+            self.GAME_CLEAR_STATUS_NO_PLAY: 'NO PLAY',
+            self.GAME_CLEAR_STATUS_FAILED: 'FAILED',
+            self.GAME_CLEAR_STATUS_ASSIST_CLEAR: 'ASSIST CLEAR',
+            self.GAME_CLEAR_STATUS_EASY_CLEAR: 'EASY CLEAR',
+            self.GAME_CLEAR_STATUS_CLEAR: 'CLEAR',
+            self.GAME_CLEAR_STATUS_HARD_CLEAR: 'HARD CLEAR',
+            self.GAME_CLEAR_STATUS_EX_HARD_CLEAR: 'EX HARD CLEAR',
+            self.GAME_CLEAR_STATUS_FULL_COMBO: 'FULL COMBO',
+        }
+        # first we'll grab the data from the packet
+        # did = request.child_value('did')
+        # rid = request.child_value('rid')
+        name = request.child_value('name')
+        # qpro_hair = request.child_value('qpro_hair')
+        # qpro_head = request.child_value('qpro_head')
+        # qpro_body = request.child_value('qpro_body')
+        # qpro_hand = request.child_value('qpro_hand')
+        music_id = request.child_value('music_id')
+        class_id = request.child_value('class_id')
+        # no_save = request.child_value('no_save')
+        # is_couple = request.child_value('is_couple')
+        # target_graph = request.child_value('target_graph')
+        target_exscore = request.child_value('target_exscore')
+        # pacemaker = request.child_value('pacemaker')
+        best_clear = request.child_value('best_clear')
+        # best_djlevel = request.child_value('best_djlevel')
+        # best_exscore = request.child_value('best_exscore')
+        # best_misscount = request.child_value('best_misscount')
+        now_clear = request.child_value('now_clear')
+        # now_djlevel = request.child_value('now_djlevel')
+        now_exscore = request.child_value('now_exscore')
+        # now_misscount = request.child_value('now_misscount')
+        now_pgreat = request.child_value('now_pgreat')
+        now_great = request.child_value('now_great')
+        now_good = request.child_value('now_good')
+        now_bad = request.child_value('now_bad')
+        now_poor = request.child_value('now_poor')
+        now_combo = request.child_value('now_combo')
+        now_fast = request.child_value('now_fast')
+        now_slow = request.child_value('now_slow')
+        best_clear_string = clear_map.get(best_clear, 'NO PLAY')
+        now_clear_string = clear_map.get(now_clear, 'NO PLAY')
+        # let's get the song info first
+        song = self.data.local.music.get_song(self.game, self.music_version, music_id, class_id)
+        notecount = song.data.get('notecount', 0)
+        # Construct the dictionary for the broadcast
+        card_data = {
```

Comment:
Ideally the left hand side of this dictionary should be constants, defined in the triggers class or common. Maybe for a future PR though? This looks brittle.

Reply:
Can we at least put a TODO in here to do that work? I should probably have requested that...

Reply:
https://github.com/DragonMinded/bemaniutils/issues/27 I created a task so feel free to add a todo at your discretion.